core/src/main/scala/cats/data/IndexedReaderWriterStateT.scala

```diff
@@ -821,8 +836,43 @@ private trait IRWSTSemigroupK1[F[_], E, L, SA, SB] extends SemigroupK[IndexedRea
     }
 }
 
-private trait RWSTAlternative1[F[_], E, L, S]
+private trait RWSTNonEmptyAlternative1[F[_], E, L, S]
```

Comment:
Sorry if I missed it. Did we add law tests for this instance?

Reply:
Yep, no worries, here they are:
https://github.com/typelevel/cats/blob/f2d871cdbb6bcb5ee2c25ebfadab90b84329b9bf/laws/src/main/scala/cats/laws/NonEmptyAlternativeLaws.scala#L27
https://github.com/typelevel/cats/blob/f2d871cdbb6bcb5ee2c25ebfadab90b84329b9bf/laws/src/main/scala/cats/laws/discipline/NonEmptyAlternativeTests.scala#L30

Reply:
Thanks, I mean specifically for the `IndexedReaderWriterStateT` instance of `NonEmptyAlternativeLaws`.

Reply:
Accordingly, `AlternativeLaws` and `AlternativeTests` were changed to extend `NonEmptyAlternative*`:
https://github.com/typelevel/cats/blob/f2d871cdbb6bcb5ee2c25ebfadab90b84329b9bf/laws/src/main/scala/cats/laws/AlternativeLaws.scala#L28
https://github.com/typelevel/cats/blob/f2d871cdbb6bcb5ee2c25ebfadab90b84329b9bf/laws/src/main/scala/cats/laws/discipline/AlternativeTests.scala#L30

Reply:
@armanbilge is it what you're asking about?
https://github.com/typelevel/cats/blob/f2d871cdbb6bcb5ee2c25ebfadab90b84329b9bf/tests/shared/src/test/scala/cats/tests/IndexedReaderWriterStateTSuite.scala#L451-L466

Reply:
That part of tests was not changed, though, iirc – it included tests for NonEmptyAlternativeLaws automatically due to the inheritance.

Reply:
I see, thanks. I was asking, because for the other instances you added tests using `ListWrapper.nonEmptyAlternative` in addition to existing tests using the `ListWrapper.alternative`, as far as I can tell.

Reply:
Oh, I see... That's a fair concern.. And honestly, I don't remember why. There are three causes possible:
1. I simply forgot to add it.
2. I lost it after a series of successive rebases.
3. Most likely, little green men came and stole that piece of code from the repo.

But I think, despite of the cause, it makes sense to return it back.